superset/annotation_layers/annotations/commands/delete.py

```diff
@@ -38,6 +38,8 @@ def __init__(self, model_id: int):
 
     def run(self) -> Model:
         self.validate()
+        assert self._model
```

Comment:
This was needed because now Mypy correctly identified that on line #44 `self._model` could have been `None` which didn't adhere to the base class definition.

Reply:
I have a draft SIP recommending a slight refactor of the commands which address this problem.

Reply:
@betodealmeida I’ve been thinking about the DAOs, commands and validation a bit recently and wonder whether we should adopt the “ask for forgiveness” try approach more often which relies less on Python validation and more on the database schema (foreign keys, uniqueness constraints, etc.) for validation, i.e., the DAO or command would fail if invalid.

The benefits are that we would reduce the Python code footprint and prevent possible race conditions.